jujugui/static/gui/src/app/app.js

```diff
@@ -1892,9 +1892,14 @@ YUI.add('juju-gui', function(Y) {
       @param {String} uuid The environment UUID where to switch to.
       @param {String} username The username for the new environment.
       @param {String} password The password for the new environment.
+      @param {Boolean} reconnect Whether to reconnect to a new environment.
     */
-    switchEnv: function(socketUrl, username, password) {
-      console.log('switching to new socket URL:', socketUrl);
+    switchEnv: function(socketUrl, username, password, reconnect=!!socketUrl) {
```

Comment:
It doesn't look like you ever pass in a reconnect param so maybe that should be moved to a scoped var rather than assigning the var using the param defaults.